src/main/thrift/parquet.thrift

```diff
@@ -174,7 +174,18 @@ enum ConvertedType {
    * particular timezone or date.
    */
   INTERVAL = 21;
-  
+
+  /**
+   * A Union type
+   * This type annotates data stored as a Group.
+   * This shows the intent to have heterogenous types under the same field name.
+   * The names of the fields in the annotated Group are not important in such a case.
+   * All fields of the Group must be optional and exactly one is defined for each instance of the group.
+   * If more than one is defined the behavior is undefined and may changed depending on the projection applied.
+   * A Union can not contain null but can be null itself if it's an optional field.
```

Comment:
Could this say "Union groups that are required must contain at least one non-null field. Union groups that are optional can be null, which is used to encode the case where none of the branches of the union are non-null."? I want to be very clear how the union group's definition level is used to encode a null value, and how implementations should use that bit to distinguish between a null union value and a missing non-null branch.